LCM/scripts/PerformRequiredConfigurationChecks.py

```diff
@@ -28,26 +36,80 @@ def main():
         raise
 
 def run_perform_required_configuration_checks():
+
+    dsc_sysconfdir = join(helperlib.CONFIG_SYSCONFDIR, helperlib.CONFIG_SYSCONFDIR_DSC)
     omicli_path = join(helperlib.CONFIG_BINDIR, 'omicli')
+    dsc_host_base_path = '/opt/dsc'
+    dsc_host_path = join(dsc_host_base_path, 'bin/dsc_host')
+    dsc_host_output_path = join(dsc_host_base_path, 'output')
+    dsc_host_lock_path = join(dsc_host_base_path, 'dsc_host_lock')
+    dsc_host_switch_path = join(dsc_host_base_path, 'dsc_host_ready')
+
+    if ("omsconfig" in helperlib.DSC_SCRIPT_PATH):
+        write_omsconfig_host_event(pathToCurrentScript, isfile(dsc_host_switch_path))
+
+    if ("omsconfig" in helperlib.DSC_SCRIPT_PATH) and (isfile(dsc_host_switch_path)):
+        use_omsconfig_host = True
+    else:
+        use_omsconfig_host = False
 
     parameters = []
-    parameters.append(omicli_path)
-    parameters.append("iv")
-    parameters.append(helperlib.DSC_NAMESPACE)
-    parameters.append("{")
-    parameters.append("MSFT_DSCLocalConfigurationManager")
-    parameters.append("}")
-    parameters.append("PerformRequiredConfigurationChecks")
-    parameters.append("{")
-    parameters.append("Flags")
-    parameters.append("1")
-    parameters.append("}")
+    if use_omsconfig_host:
+        parameters.append(dsc_host_path)
+        parameters.append(dsc_host_output_path)
+        parameters.append("PerformRequiredConfigurationChecks")
+        parameters.append("1")
+    else:
+        parameters.append(omicli_path)
+        parameters.append("iv")
+        parameters.append(helperlib.DSC_NAMESPACE)
+        parameters.append("{")
+        parameters.append("MSFT_DSCLocalConfigurationManager")
+        parameters.append("}")
+        parameters.append("PerformRequiredConfigurationChecks")
+        parameters.append("{")
+        parameters.append("Flags")
+        parameters.append("1")
+        parameters.append("}")
 
     # Save the starting timestamp without milliseconds
     startDateTime = operationStatusUtility.get_current_time_no_ms()
 
-    process = Popen(parameters, stdout = PIPE, stderr = PIPE)
-    stdout, stderr = process.communicate()
+    if use_omsconfig_host:
+        try:
+            # Open the dsc host lock file. This also creates a file if it does not exist
+            dschostlock_filehandle = open(dsc_host_lock_path, 'w')
+            print("Opened the dsc host lock file at the path '" + dsc_host_lock_path + "'")
+            
+            dschostlock_acquired = False
+
+            # Acquire dsc host file lock
+            for retry in range(10):
+                try:
+                    flock(dschostlock_filehandle, LOCK_EX | LOCK_NB)
+                    dschostlock_acquired = True
+                    break
+                except IOError:
+                    write_omsconfig_host_log(pathToCurrentScript, 'dsc_host lock file not acquired. retry (#' + str(retry) + ') after 60 seconds...')
+                    time.sleep(60)
```

Comment:
Need to brainstrom if this logic will not get us into a situation where one solution is not getting a chance to execute.